autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py

```diff
@@ -573,15 +573,10 @@ def _get_types_of_features(self, df, skew_threshold, embed_min_categories, use_n
         if self.types_of_features is not None:
             Warning("Attempting to _get_types_of_features for TabularNeuralNetModel, but previously already did this.")
 
-        # TODO: Consider setting use_ngram_features=True by default once performance is improved
-        if not use_ngram_features:
-            vectorizers_featnames = self.__get_feature_type_if_present('vectorizers')
-            nlp_featnames = self.__get_feature_type_if_present('nlp')
-            self.feature_types_metadata['int'] = [feature for feature in self.__get_feature_type_if_present('int') if feature not in vectorizers_featnames]
-            self.feature_types_metadata['object'] = [feature for feature in self.__get_feature_type_if_present('object') if feature not in nlp_featnames]
+        feature_types = self.feature_types_metadata.feature_types_raw
 
-        categorical_featnames = self.__get_feature_type_if_present('object') + self.__get_feature_type_if_present('bool')
-        continuous_featnames = self.__get_feature_type_if_present('float') + self.__get_feature_type_if_present('int') + self.__get_feature_type_if_present('datetime')
+        categorical_featnames = feature_types['category'] + feature_types['object'] + feature_types['bool']
+        continuous_featnames = feature_types['float'] + feature_types['int']  # + self.__get_feature_type_if_present('datetime')
```

Comment:
why ignore datetime?

Reply:
Already in float/int. The way metadata is structured has changed, raw feature_types are the actual raw DataFrame dtypes (after combining into families). Raw has no context of the special meaning behind features, that is what Special is for.